### ISSUE TITLE
desi_resubmit_queue_failures job chaining

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1558]
Submitting a holiday ticket while the issue is fresh on my mind, but ok to wait until next week to address it --

While processing f4, night 20210509 was in a state where the processing table listed 3 arc jobs were "COMPLETED", 2 as "FAILED", and 1 as TIMEOUT.  I unfortunately did not cache this state before proceeding with:
```
desi_resubmit_queue_failures -n 20210509
```
This resubmitted the "TIMEOUT" job but not the "FAILED" ones because "FAILED" is not in the default list of "--resub-states" options, perhaps because "FAILED" is more likely due to a genuine algorithmic failure than a known slurm/nersc problem.  Perhaps FAILED should be in the list of auto-resubmit states.

Side note: In this case it isn't clear why they failed; e.g. arc-20210509-00087766-a0123456789-52224893.log has "slurmstepd: error: *** STEP 52224893.0 ON nid02537 CANCELLED AT 2021-12-25T08:04:57 ***" (about 13 minutes into a 45 minute job) without a clear reason why.  I'll resubmit to test that too.

I was surprised however that it proceeded to submit the chain of psfnight, flats, and nightlyflat, starting with the psfnight depending only on that one resubmitted arc even though there were still two FAILED exposures that the previous psfnight job had depended upon.  Maybe this is purposeful, thinking that we are supposed to cleanup FAILED as best we can before giving up and resubmitting the rest.  Or maybe it should have blocked on the FAILED (in which case I'm not sure how I would tell the pipeline to resubmit and proceed anyway if I couldn't get beyond the failure but wanted it to proceed with whatever it has).  Feature or bug?

Question: IIUC, when doing a prod with desi_run_night, the processing table is written but then never updated again unless that night is resubmitted, i.e. everything remains in the "PENDING" or "SUBMITTED" state.  Is there a command to force an update of the processing table without resubmitting jobs to get the jobs flagged as FAILED, TIMEOUT, CANCELED, etc.?

@akremin comments?

**Comment by sbailey** [comment]
Related to resubmitting jobs in the FAILED state, some other examples of f4 jobs that have slurm status FAILED:

arc-20210212-00075558-a0123456789-52189930.log
```
Starting at Wed Dec 22 13:53:52 PST 2021
Running srun -N 10 -n 301 -c 9 desi_proc --traceshift --cameras a0123456789 -n 20210212 -e 75558 --mpi --timingfile /global/cfs/cdirs/desi/spectro/redux/f4/run/scripts/night/20210212/arc-20210212-00075558-a0123456789-timing-52189930.json
srun: error: Unable to confirm allocation for job 52189930: Socket timed out on send/recv operation
srun: Check SLURM_JOB_ID environment variable. Expired or invalid job 52189930
FAILED: done at Wed Dec 22 13:55:34 PST 2021
```

arc-20210203-00074365-a0123456789-52189717.log:
```
...
  File "skimage/morphology/_max_tree.pyx", line 1, in init skimage.morphology._max_tree
    from . import _max_tree
  File "skimage/morphology/_max_tree.pyx", line 1, in init skimage.morphology._max_tree
RuntimeError: Unable to initialize pickling for skimage.morphology._max_tree.array
RuntimeError: Unable to initialize pickling for skimage.morphology._max_tree.memoryview
srun: error: nid06573: tasks 14,28: Exited with exit code 1
srun: launch/slurm: _step_signal: Terminating StepId=52189717.0
slurmstepd: error: *** STEP 52189717.0 ON nid06573 CANCELLED AT 2021-12-22T09:57:15 ***
```

arc-20210202-00074229-a0123456789-52189691.log
```
    from desispec.qproc.qfiberflat import qproc_apply_fiberflat
ImportError: cannot import name 'qproc_apply_fiberflat' from 'desispec.qproc.qfiberflat' (/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/master/py/desispec/qproc/qfiberflat.py)
    from desispec.qproc.qfiberflat import qproc_apply_fiberflat
ImportError: cannot import name 'qproc_apply_fiberflat' from 'desispec.qproc.qfiberflat' (/global/common/software/desi/cori/desiconda/20211217-2.0.0/code/desispec/master/py/desispec/qproc/qfiberflat.py)
srun: error: nid06573: tasks 17,29: Exited with exit code 1
srun: launch/slurm: _step_signal: Terminating StepId=52189691.0
slurmstepd: error: *** STEP 52189691.0 ON nid06573 CANCELLED AT 2021-12-22T09:54:07 ***
```

These all appear to be one-off problems due to slurm or /global/common/software hiccups and would be cases that we would want to auto-resubmit, i.e. I think we should add "FAILED" to the cases that auto-resubmit, even if that would resubmit (and re-fail) on cases that were genuinely our fault.

That still leaves the original question of whether the re-submission should continue with the downstream jobs even if they originally depended upon jobs that are not resubmitted and not already "COMPLETED".

**Comment by sbailey** [comment]
fixed by #1602.
